src/high-level/util.lisp

```diff
@@ -0,0 +1,99 @@
+;;;; util.lisp
+;;;;
+;;;; Author: Cole Scott
+
+(in-package #:magicl)
+
+(defconstant +float-comparison-threshold-loose+   1d-5)
+(defconstant +float-comparison-threshold-strict+  5d-8)
+(defconstant +double-comparison-threshold-loose+  1d-5)
+(defconstant +double-comparison-threshold-strict+ 5d-11)
```

Comment:
Please write these as multiples of, respectively, `single-float-epsilon` and `double-float-epsilon`.

I know that we have these floating point comparisons all over the place in our code base but they're generally not meaningful. We should establish "equality" of floating point numbers based on relative error instead of absolute error, as we currently do. This is out of scope for this PR, though. I'm just putting it out there.

Reply:
It now stops making a distinction between strict and loose thresholds and instead uses `single-float-epsilon` and `double-float-epsilon`. This should hold us over until relative error-based comparisons can be added